### PR TITLE
Task-outcomes card migration to Angular

### PR DESCRIPTION
**Pull request by almuntasir**
# Description

I have added the typescript and angular component to migrate the task-outcomes card. I did some changes to the code logic as I sent the task object to the component and handle getting data in the component code.      

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x]  New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

I have tested changes locally and I got the expected learning outcomes based on the given task.

# Checklist:

- [x]  My code follows the style guidelines of this project
- [x]  I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x]  My changes generate no new warnings
- [x]  I have added tests that prove my fix is effective or that my feature works
- [x]  New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

If you have any questions, please contact @macite or @jakerenzella.
task-outomes1

### CODE REVIEW

**Comment by jakerenzella** [comment]
Pull request description needs to be completed, please read the template for guidance and replace with the appropriate content.

**Comment by almuntasir** [comment]
Hi Jake, 

I updated the pull request, please consider the task-outcomes-card and give me your feedback. Regards.

File: src/app/projects/states/dashboard/directives/task-dashboard/directives/task-outcomes-card/task-outcomes-card.component.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          <!-- <p>
+            <h5>Outcome Description</h5>
+            <div class="markdown-to-html" 
+            ng-bind-html="alignment.ilo.description | markdown"></div>
+          </p> -->
+          <!-- <p>
+            <h5>Rationale</h5>
+            <div class="markdown-to-html" 
+            ng-bind-html="alignment.description | markdown"></div>
+          </p> -->
```
Comment (jakerenzella) [inline src/app/projects/states/dashboard/directives/task-dashboard/directives/task-outcomes-card/task-outcomes-card.component.html:40]: What is this commented block?

File: src/app/projects/states/dashboard/directives/task-dashboard/directives/task-outcomes-card/task-outcomes-card.component.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  // that maps all of the form controls that this form consists of.
+  constructor() {
+    
+  }
+  ngOnInit(): void {
+    this.unit= this.task.unit();
+    this.taskDef= this.task.definition;
+    this.alignments = this.unit.staffAlignmentsForTaskDefinition(this.taskDef);   
+  }  
+  //constructor(@Inject(outcomeService) private outcomes: any) {}
```
Comment (jakerenzella) [inline src/app/projects/states/dashboard/directives/task-dashboard/directives/task-outcomes-card/task-outcomes-card.component.ts:26]: ?

File: src/app/projects/states/dashboard/directives/task-dashboard/directives/task-outcomes-card/task-outcomes-card.component.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  ngOnInit(): void {
+    this.unit= this.task.unit();
+    this.taskDef= this.task.definition;
+    this.alignments = this.unit.staffAlignmentsForTaskDefinition(this.taskDef);   
+  }  
+  //constructor(@Inject(outcomeService) private outcomes: any) {}   
+  ngOnChanges(changes: SimpleChanges ) {   
+    let change = changes['taskDef.id'];
+    let curVal  = JSON.stringify(change.currentValue);
+    let prevVal = JSON.stringify(change.previousValue);
```
Comment (jakerenzella) [inline src/app/projects/states/dashboard/directives/task-dashboard/directives/task-outcomes-card/task-outcomes-card.component.ts:30]: What are we doing with these?

File: src/app/projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.tpl.html [hunk 4]
```diff
@@ -56,5 +56,26 @@ <h4>
...
+    <thead> 
+        <tr> 
+          <th> Comment </th>
+          <th> Type </th>
+          <th> Marker </th>
+          <th> Read Date </th>
+        </tr>                                             
+    </thead>
+    <tbody>
+      <tr ng-repeat="comment in task.comments">
```
Comment (jakerenzella) [inline src/app/projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.tpl.html:71]: this is not Angular

**Review by jakerenzella — CHANGES_REQUESTED** [review]

**Pull request closed without merging**